src/ucs/arch/ppc64/cpu.h

```diff
@@ -25,9 +25,10 @@ BEGIN_C_DECLS
 #define ucs_memory_bus_fence()        asm volatile ("sync"::: "memory")
 #define ucs_memory_bus_store_fence()  ucs_memory_bus_fence()
 #define ucs_memory_bus_load_fence()   ucs_memory_bus_fence()
-#define ucs_memory_cpu_fence()        ucs_memory_bus_fence()
-#define ucs_memory_cpu_store_fence()  ucs_memory_bus_fence()
-#define ucs_memory_cpu_load_fence()   ucs_memory_bus_fence()
+#define ucs_memory_cpu_fence()        asm volatile ("lwsync"::: "memory")
+#define ucs_memory_cpu_store_fence()  asm volatile ("lwsync"::: "memory")
+#define ucs_memory_cpu_load_fence()   asm volatile ("lwsync"::: "memory")
+#define ucs_cpu_instruction_fence()   asm volatile ("isync"::: "memory")
```

Comment:
maybe let's remove ucs_cpu_instruction_fence() and push "isync" inside ucs_memory_cpu_load_fence() in addition to "lwsync"?